Regenerate wilms notebooks: Code updates

sjspielman opened the pull request:
This is the first of 2 PRs to address #906 (but might be more if GitHub UI gets mad) to update notebooks in the `cell-type-wilms-tumor-06`. This PR covers a smidge of reorg and code changes that cropped up while regenerating notebooks with the 2024-11-25 data release.

1. I changed a few things to better handle dealing with the exploratory CNV notebooks which are not going to get regenerated:
    - I created a new directory `cnv-exploratory-notebooks` for them and their output to live in. I also had to update the path accordingly in `scripts/explore-cnv-methods.sh`.
    - I added a flag to the `00_run_workflow.sh` script to specifically handle logic for regenerating these notebooks. Even though we will want to run other exploratory steps in the future, we almost certainly won't want to run these!
2. I moved `00b_characterize_fetal_kidney_reference_Stewart.Rmd` out of notebook_template and into notebook, since it is not a template. 
    - I updated `00_run_workflow.sh` with the right new path, and also I also renamed that script variable `notebook_output_dir` --> `notebook_dir`, since the directory is named "notebook". 
3. I realized (WOMP) that `scripts/06b_build-normal_reference.R` was missing from the workflow when running the workflow through! The script also needed a Seurat library load which snuck past during review because this module was mostly not being run in CI. The script is now run, loads Seurat, and was styled by pre-commit.
4. The 03 clustering notebook had some Seurat-related bugs which appeared with the new data release: AddModuleScore doesn't work with defaults when there are fewer cells (<500), so I updates some spots to get this code running. This is now relevant because at least one sample lost about 20-30 cells.
5. README updates along the way

Please let me know what I can clarify!!

allyhawkins reviewed (commented):
I left a few comments about the approach here. Based on https://github.com/AlexsLemonade/OpenScPCA-analysis/issues/906#issuecomment-2489649689, I was thinking we would completely remove generating these notebooks from the workflow. Since we have a script that is already used to run two scripts and generate the notebooks then I don't think it needs to be present in the main workflow. 

The other comment is about how we organize the exploratory analysis. Since the `supplemental-notebooks` exists, then we should put everything that's exploratory/ not run in the workflow in that folder.

sjspielman commented on analyses/cell-type-wilms-tumor-06/scripts/README.md:
Boo, GitHub.
This was changed because `06b_build-normal_reference.R` wasn't documented, and I had a brief moment of recreational revision to make 06 scripts listed after 05 scripts.

Here's the diff which renders for me:
![Screenshot 2024-11-27 at 11 15 39 AM](https://github.com/user-attachments/assets/162bbc4c-e2f9-4c6a-8141-483c4eb01ccd)

sjspielman commented on analyses/cell-type-wilms-tumor-06/00_run_workflow.sh:
```diff
@@ -134,17 +142,23 @@ if [[ $RUN_EXPLORATORY -eq 1 ]]; then
-  # This script calls scripts/05_copyKAT.R and scripts/06_infercnv.R
-  # By default, copyKAT as called by this script uses 32 cores
-  THREADS=${THREADS} TESTING=${TESTING} ./scripts/explore-cnv-methods.sh
+  # These steps are only run if RUN_CNV_EXPLORATORY is true
+  if [[ $RUN_CNV_EXPLORATORY -eq 1 ]]; then
+    # Create the pooled normal reference for inferCNV
+    Rscript scripts/06b_build-normal_reference.R
+
+    # Run infercnv and copykat for a selection of samples
+    # This script calls scripts/05_copyKAT.R and scripts/06_infercnv.R and associated exploratory CNV notebooks in `cnv-exploratory-notebooks/`
+    # By default, copyKAT as called by this script uses 32 cores
+    THREADS=${THREADS} TESTING=${TESTING} ./scripts/explore-cnv-methods.sh
+  fi
```
No complaints from me.

sjspielman commented:
Fresh changes:

- All the "not part of this workflow" notebooks are now in `supplemental_notebooks` (check out that underscore! This is most of the new diffs - renaming this directory to have an underscore)
  - This includes the cnv exploration notebooks, and an exploratory notebook which was used to classify one of the label transfer references, but indeed is wholly unused in cell typing itself. 
- `marker-sets` --> `marker_sets`, for more consistency. You can have all `_`, but not all `-`.  
- I updated the workflow script to not run anything that lives in `supplemental_notebooks`, and made it clearer that the `RUN_EXPLORATORY` flag can be used to regenerate all notebooks that live in the repo.
- README updates accordingly.

I think I recommend looking directly at the two commits I added since your last review to review this, since it will make identifying the specific changes easier. Again, please let me know where I can clarify!

sjspielman commented on analyses/cell-type-wilms-tumor-06/README.md:
```diff
@@ -24,6 +24,19 @@ The analysis is/will be divided as the following:
+
+- `scripts` contain analysis scripts used in the module workflow.
+See `scripts/README.md` for more information
+- `notebook_template` contains R Markdown notebooks meant to be run as a template across module samples
```
It should now all be `_` :)

allyhawkins reviewed (approved):
New changes look good 👍
